tests/unit/cache_api_test.py

```diff
@@ -41,6 +41,23 @@ def test_expiration(self):
         self.assertIsNone(self.cache.get('foo'))
         self.assertIsNone(self.cache.get('foo', namespace='bar'))
 
```

Comment:
Does datastore overwrite existing key/value pairs? I would imagine it does and the behavior would look like this:

```suggestion

    def test_multiple_adds_single_delete(self):
        self.assertTrue(self.cache.add('foo', 0))
        self.assertTrue(self.cache.add('foo', 42))
        self.assertEqual(self.cache.get('foo'), 42)
        self.assertIsNone(self.cache.delete('foo'))
        self.assertIsNone(self.cache.get('foo'))

```

(We don't need to test the behavior of datastore right?)

Reply:
I'm not sure what datastore does -- is there something you're concerned about that's impacted by whether or not it overwrites existing values? I don't think we necessarily need this particular test, but I think I'm going to try and add an an unlinking test that's backed by the datastore emulator

Reply:
yeah I was just wondering if there's any reason to believe that the above test would fail (like if datastore stored both of the `add` entries instead of overwriting the existing one... and I just realized that this is probably using fake_cache_api.py. Having a test using the datastore emulator sounds like a good idea!